ivy/functional/ivy/nest.py

```diff
@@ -788,36 +788,92 @@ def nested_argwhere(
 
 @handle_exceptions
 def all_nested_indices(
-    nest: Iterable,
-    include_nests: bool = False,
-    _index: Optional[List] = None,
-    _base: bool = True,
-    extra_nest_types: Optional[Union[type, Tuple[type]]] = None,
-) -> Union[Iterable, bool]:
-    """Returns indices of all the elements in nest
+    nest: Optional[Iterable] = None,
```

Comment:
I would say it's better to use `Union[List, Tuple, Dict, Array, NativeArray, Container]`